scripts/esbuild-commands.js

```diff
@@ -36,7 +36,7 @@ const createEsbuildTask = (esbuildOptions = {}, otherOptions = {}) => {
         if (!fileEventName || !filePath) {
           throw new Error('Arguments `fileEventName` and `filePath` must be provided when rerunning this task.')
         }
-        await state.result.rebuild()
+        state.result = await state.result.rebuild()
```

Comment:
What's going on here?

@snowteamer could you review this change since this is code you wrote?

Reply:
Things have changed in recent esbuild versions, but in v0.14.47 according to https://github.com/evanw/esbuild/blob/v0.14.47/lib/shared/types.ts the `rebuild()` method of `BuildResult` objects didn't return a new ˘BuildResult` object, but an object implementing `BuildInvalidate` instead, so I'm not sure about this change

Reply:
@taoeffect , @snowteamer 
This change was made in an attempt to fix a bug that I constantly had where esbuild fails to perform rebuild in response to a modification in any pure `.js` files in the project folder. The Browsersync watcher did successfully detect a new file addition or updates to existing js files but execution of the 'rebuild()' command that runs as a result of it didn't do anything so I had to keep terminating & re-launching dev-server (the same work for `.vue`, `.scss` files had always worked well though).

After I made this change, everything worked well and from what I've observed, it didn't cause a problem to GI code-base :)

Reply:
@SebinSong how did you figure out to do this? Did you find a page that recommended doing this online somewhere? If so, could you add a comment with a link to it?

Reply:
@taoeffect I actually gave this one a try by myself and it worked. And according to the link @snowteamer attached above, it doesn't violate the type definition. (`rebuild()` does return `BuildResult`)

![image](https://user-images.githubusercontent.com/17641213/231596697-11a6c705-58d4-4ff9-8cca-39619e6c75d5.png)

I will leave a comment for this btw.

Reply:
OK, but I'm just trying to figure out, what gave you the idea to do this in the first place? Did you decide to look into the esbuild source yourself and come to the conclusion that this might help? I assume you didn't just randomly start changing things... right? :)

Reply:
@taoeffect I did spend some time on studying esbuild docs (I wanted to understand at least how it works instead of just copy&pasting the existing`Gruntfile.js`) and esbuild-related modules in our source-code Snowteamer created when I worked on setting up build-steps for chelonia dashboard and while trying to figure out this problem. and that particular part of `esbuild-commands.js` file looked suspicious. it did run `rebuild()` method but somehow doesn't refresh the result of the initial startup build, so then thought I would try it this way.

Reply:
Or I can put it back if this seems problematic.

Reply:
I mean, if it works it works! Just want to make sure we aren't accidentally fixing something while also making something worse. Thanks for the explanation! You're right, we can keep it and if it causes problems we can always remove it.